include/open62541pp/types.hpp

```diff
@@ -888,59 +961,99 @@ class Variant : public TypeWrapper<UA_Variant, UA_TYPES_VARIANT> {
 public:
     using TypeWrapper::TypeWrapper;  // inherit constructors
 
+    // Universal constructors with default copy policy
+
+    // TODO:
+    // Allow implicit constructor?
+    template <typename T, typename X = std::enable_if_t<!std::is_same_v<T, Variant>, void>>
+    explicit Variant(T&& value) {
+        setValueCopy(std::forward<T>(value));
```

Comment:
If you look at the new universal constructors and the `setXyz` API side-by-side, they don't seem to match very well:

- The constructors copy by default, and you opt into referencing the data by using a reference tag
- The `setXyz` API references the data, and it feels like you opt into copying by using `setXyzCopy`

It feels confusing to have this kind of mismatch in the API. We could adapt `setValue` in a way that resembles the new constructors more closely: We could either turn around the API names to `setValue / setValueReference` or use a single `setValue` API with tag dispatch in the same way we do in the constructor.

From API perspective I find using a single `setValue` API with the same tag dispatch mechanism the most appealing, though that does not match well with the scalar and array get/set functions 😅 

What do you think? @lukasberbuer

Reply:
To be honest, it doesn't bother me that much having a `ReferenceTag` and the suffix `Copy` for setters/getters, but I get your point.
On the one hand, I would like to stick to `set*Copy` and `get*Copy` to be close to the open62541 API. On the other hand, copy should be the default (safe) choice without any constructor tag.

Implementing a `setValue` with a tag and forwarding it either to `setScalar(Copy)` or `setArray(Copy)` should be easy to implement and could be done outside of this library.
My biggest concern is that users who are not so familiar with templates will be put off by such API choices.

Reply:
Changed API to `setValue` / `setValueCopy`

Reply:
I'm not sure if we had discussed your idea [using an assignment operator](https://github.com/open62541pp/open62541pp/pull/384#issuecomment-2434568042). I suppose it's probably not a good option if you prefer to go with `setValue `/ `setValueCopy`, and it would also require a reference tag?

Reply:
I'm not sure if we had discussed [your idea about using assignment operators](https://github.com/open62541pp/open62541pp/pull/384#issuecomment-2434568042). I suppose it's not a good option if you want to go with `setValue` / `setValueCopy`, and the assignment operators would also require a reference tag?